test/test_type_promotion.py

```diff
@@ -268,6 +268,22 @@ def test_comparison_ops_with_type_promotion(self):
                     self.assertTrue(t1.dtype == dt1)
                     self.assertTrue(t2.dtype == dt2)
 
+                    t1 = torch.tensor(val1, dtype=dt1, device=device)
```

Comment:
It would be helpful if you added a comment here with something along the lines of "we are testing that comparing a zero dim tensor with another zero dim tensor has type promotion behavior"

Reply:
Added comment